### DISCUSSION TITLE
word-level-timestamps calculates promp incorrectly and disregards condition_on_previous_text (bug)

### DISCUSSION THREAD

**Discussion by jmward01** [openai/whisper#995] [General]
Unless I am missing something, the calculation for prompt always includes at least the previous segment tokens even if condition_on_previous_text = False since prompt_reset_since is calculated before the tokens from the current segment are added. This behavior is only in the word-level-timestamps branch currently. The fix is to just move this calculation to the end like the main branch has it.

https://github.com/openai/whisper/blob/8eb29c3ef10559910cbee47b1baedefd8388458a/whisper/transcribe.py#L259

**Comment by clementabary** [comment]
Yep, I found out while fine-tuning Whisper with timestamps but without ever giving it previous tokens. While debugging the transcribe function, it seems that with `condition_on_previous_text=False` it still gives it the transcription from the last segment, so technically it does not feel like "not conditioned"
